lib/plugins/simpleRetry.js

```diff
@@ -17,30 +21,70 @@ var simpleRetry = function(worker, func, queue, job, args, options){
     self.queueObject = self.worker;
   }
 
-  self.sleep = 1000;
-  if(self.options.sleep){ self.sleep = self.options.sleep; }
+  if (self.args) {
+    jobHash = crypto.createHash('sha256').update(JSON.stringify(self.args)).digest('hex');
+    self.retryKey = self.queueObject.connection.key('retryTimes', jobHash);
+  }
 };
 
 ////////////////////
 // PLUGIN METHODS //
 ////////////////////
 
+simpleRetry.prototype.updateRetryTimes = function(callback) {
+  var self = this;
+
+  self.queueObject.connection.redis.incr(self.retryKey, (function(err, result) {
+    if (err) { return callback(err); }
+
+    self.queueObject.connection.redis.expire(self.retryKey, self.options.retryInterval[self.options.retryInterval.length - 1] * 2, function(err) {
+      if (err) {
+        callback(err);
+      } else if (result > self.options.retryInterval.length) {
+        callback(new Error('Retry limit exceeded'));
```

Comment:
I think its probably important to keep the original stack trace from the failure (so we'll know that after many attempts, line X is the source of the problem).  If you want to denote that this plugin was used, you could modify `error.message = '(Resque Retry Max Attempts Reached) -> 'error.message` or something like that.  This would preserver the original error message and stack.